File: pyprojects/sentimental-readability/readability.py
# To make readability in python
from cs50 import get_string

# For text
text = get_string("Text: ")

# To count number of words, letters and sentences
letters = 0
words = 1
sentences = 0

# For numbers of letters
for i in range(0, len(text), 1):
    if (text[i].isalpha()) == True:
        letters += 1
    # For numbers of words
    if (text[i].isspace()) == True:
        words += 1
    # For numbers of sentences
    if text[i] == '.' or text[i] == '?' or text[i] == '!':
        sentences += 1

# To calculate the text
Lett = letters / words * 100
Sent = sentences / words * 100

# Convert into float
L = float(Lett)
S = float(Sent)

index = round(0.0588 * L - 0.296 * S - 15.8)

# To print the Grade no. of the text
if index < 1:
    print("Before Grade 1")
elif index > 16:
    print("Grade 16+")
else:
    print(f"Grade {index}")

File: cprojects/readability/readability.c
#include <cs50.h>
#include <ctype.h>
#include <stdio.h>
#include <string.h>
#include <math.h>

int main(void)
{
    string text = get_string("Text: ");

// Declaration
    int letters = 0;
    int words = 1;
    int sentences = 0;

    // Numbers of letters

    int count_letters(string text);

    for (int i = 0; i < strlen(text); i++)
    {
        if (isalpha(text[i]))
        {
            letters++;
        }

        // Numbers of words

        int count_words(string text);

        if (isspace(text[i]))
        {
            words++;
        }

        // Number of sentences

        int count_sentences(string text);

        if (text[i] == '.' || text[i] == '?' || text[i] == '!')
        {
            sentences++;
        }
    }

    // Calculation
    float L = (float) letters / (float) words * 100;
    float S = (float) sentences / (float) words * 100;

    int index = round(0.0588 * L - 0.296 * S - 15.8);

    // Print the Grades
    if (index < 1)
    {
        printf("Before Grade 1\n");
    }

    else if (index > 16)
    {
        printf("Grade 16+\n");
    }

    else
    {
        printf("Grade %i\n", index);
    }
}

